### ISSUE TITLE
[Question] Flaky test cases table - use a filter instead of launch name

### ISSUE THREAD

**Issue by CosminGramada** [reportportal/reportportal#2186]
**Describe the solution you'd like**
Currently, when creating a `Flaky test cases table (TOP-20)` widget one can only select launches in the drop-down for filtering in the widget.

It would be helpful to be able to also create the widget based on a filter.

My use case and reasoning:
- tests running from master branch (that are running on multiple repositories) should always be stable
- tests running from other branches are most of the times work in progress, where existing tests are modified, code is refactored, etc and we don't expect it to be stable (until it gets merged to master)
- using the flaky tests widget we would like to analyze only the tests that we expect to be stable (meaning those running from master branch).

I'm curious if there is maybe a workaround for the scenario that I'm describing above?

**Comment by Kozovskaya** [comment]
Hello @CosminGramada,
Could you please provide some more details for your use case?
Do you have the same launch names that are running from master branch and from other branches? If they are the same as a workaround you could make them different and build statistics for test from different branches separately.
Do you have a kind of attribute for launches that run from master? 
Do you have multiple launches with different names for master branch?  
Do you want to create the filter for the widget and built the statistics based on multiply launches that have particular attribute?

**Comment by CosminGramada** [comment]
@Kozovskaya 

> Do you have the same launch names that are running from master branch and from other branches? If they are the same as a workaround you could make them different and build statistics for test from different branches separately.

Yes, the launch names are the same (I think the other [issue that I opened as a bug](https://github.com/reportportal/reportportal/issues/2187) better paints the picture)

> Do you have a kind of attribute for launches that run from master?

Yes, I have added an attribute with information about the automation branch of the tests.

> Do you have multiple launches with different names for master branch?

Yes, quite a few, depending on the project where the tests are run.

> Do you want to create the filter for the widget and built the statistics based on multiply launches that have particular attribute?

Yes, I think this explains the setup that would help the most.

> Could you please provide some more details for your use case?

A bit of background about how we're doing CI/CD:
- we have quite a lot of microservices (each with it's own repository in Gitlab)
- we have a few test automation repositories handling different types of testing (API, frontend, etc)
- we have the ability to create testing environments on the fly where each service can be raised from an image (images being tagged with the branch name) and based on that we decide what suite of automated tests to run and from what image tag. 
As a simplified example, we can have `Service1` running from `abc` image tag (branch), and then `Service2` and `Service3` running from  `master` tag. 
From a QA perspective, there are two options: either it's just a regression so we decide to run `AutomationSuite1` from the `master` tag (branch from one of the Automation repositories) or changes are required so we need to run `AutomationSuite1` from the `abc` tag (convention is to keep to same branch name as the development service with changes).
The setup above is part of the pipeline running for `Service1` project in Gitlab.

During the flaky tests analysis we want to ignore the test runs that for `AutomationSuite1` from tag `abc` as that may be work in progress, either from QA automation point of view (new tests may be added, updated, removed, etc) and only count those running from the `master` tag.

(Sorry for the wall of text)

**Comment by CosminGramada** [comment]
@Kozovskaya 
I wanted to follow-up on this topic and see if there are any updates (even in the form of items added to the backlog).

**Comment by Kozovskaya** [comment]
Hello @CosminGramada 
Sorry for the late reply, we've discussed it with the team. And I found out that current implementation was driven by the performance issues as filtering option may lead to the analysis of huge amounts of data. In case any updates/refactoring of the widget we will take your issue into account.  
Thanks for understanding
